dockstore-common/src/main/java/io/dockstore/common/yaml/DockstoreYamlHelper.java

```diff
@@ -373,40 +382,82 @@ private static void addNewPropertyClassToQueue(String dockstoreYamlPackageName,
         }
     }
 
-    private static <T> void validate(final T validatee) throws DockstoreYamlException {
+    public static <T> void validate(final T validatee) throws DockstoreYamlException {
+        validate(validatee, true);
+    }
+
+    public static <T> void validate(final T validatee, final boolean validateEntries) throws DockstoreYamlException {
+        validate(validatee, validateEntries, ".dockstore.yml");
+    }
+
+    /**
+     * Performs constraint validation on an object, throwing an exception containing a descriptive error message if there violations.
+     * @param validatee object to validate
+     * @param validateEntries determines whether to validate the entries in a validatee representing a .dockstore.yml, should be false for other types of validatee
+     * @param validateeDescription a very short description of the validatee, used to form error messages for top-level violations
+     * @throws DockstoreYamlException an exception containing a descriptive error message that details all violations
+     */
+    public static <T> void validate(final T validatee, final boolean validateEntries, final String validateeDescription) throws DockstoreYamlException {
         final Validator validator = createValidator();
-        final Set<ConstraintViolation<T>> violations = validator.validate(validatee);
+        final Set<ConstraintViolation<T>> violations = validator.validate(validatee).stream().filter(validateEntries ? v -> true : DockstoreYamlHelper::doesNotReferenceWorkflowish).collect(Collectors.toSet());
```

Comment:
> validateEntries ? v -> true : DockstoreYamlHelper::doesNotReferenceWorkflowish

This reads the inverse of what I'm expecting, but maybe I'm missing something -- if `validateEntries` is `true`, don't you want to invoke `DockstoreYamlHelper::doesNotReferenceWorkflowish`? This code only invokes that if `validateEntries` is `false`.

Reply:
Here's one way to read it.
We've got a stream of violations.
If we want to validate the entries, include all the violations.
If we don't want to validate the entries, only include the violations that don't reference an entry.

Reply:
I should add a comment above this line that explains that it fully validates the object, then filters out the violations that reference entries if we're not interested in them.

Reply:
I think the filter expression could be rewritten as:
```
v -> validateEntries || doesNotReferenceWorkflowish(v)
```
But I personally find that harder to understand.

Reply:
> I should add a comment above this line that explains that it fully validates the object, then filters out the violations that reference entries if we're not interested in them.

+1, I would find this comment helpful

Reply:
I restructured the code to avoid the ternary op in the stream, and added some comments.